test_regress/driver.pl

```diff
@@ -1854,33 +1861,62 @@ sub _make_main {
     }
     print $fh "        ${set}fastclk = false;\n" if $self->{inputs}{fastclk};
     print $fh "        ${set}clk = false;\n" if $self->{inputs}{clk};
-    _print_advance_time($self, $fh, 10);
+    if (!$timing_loop) {
+        _print_advance_time($self, $fh, 10);
+    }
     print $fh "    }\n";
 
     my $time = $self->sc ? "sc_time_stamp()" : "contextp->time()";
 
-    print $fh "    while ((${time} < sim_time * MAIN_TIME_MULTIPLIER)\n";
+    print $fh "    while (";
+    if ($timing_loop && !$self->{inputs}{clk}) {
+        print $fh "topp->eventsPending()\n";
+    } else {
+        print $fh "(${time} < sim_time * MAIN_TIME_MULTIPLIER)\n";
+    }
     print $fh "           && !contextp->gotFinish()) {\n";
 
-    for (my $i = 0; $i < 5; $i++) {
-        my $action = 0;
-        if ($self->{inputs}{fastclk}) {
-            print $fh "        ${set}fastclk = !${set}fastclk;\n";
-            $action = 1;
+    if ($timing_loop) {
+        print $fh "        ${set}eval();\n";
+        if ($self->{trace} && !$self->sc) {
+            $fh->print("#if VM_TRACE\n");
+            $fh->print("        if (tfp) tfp->dump(contextp->time());\n");
+            $fh->print("#endif  // VM_TRACE\n");
         }
-        if ($i == 0 && $self->{inputs}{clk}) {
-            print $fh "        ${set}clk = !${set}clk;\n";
-            $action = 1;
+        if ($self->{inputs}{clk}) {
+            print $fh "        auto newTime = topp->nextTimeSlot();\n";
+            print $fh "        if (newTime == contextp->time() ||\n";
+            print $fh "            newTime / MAIN_TIME_MULTIPLIER > contextp->time() / MAIN_TIME_MULTIPLIER) {\n";
```

Comment:
This is for tests that drive the clock from main, but also use delays. I can disable timing for those.

Reply:
Some tests, like `t_delay`, assume that the clock is flipped every `#1`. But I set the timescale there to `100ns/1ns` so that the delays actually do something. `MAIN_TIME_MULTIPLIER` tells me how long `#1` is so I know when to flip the clock. What other solution do you propose?

Reply:
Ok, that's fine, when this is eventually cleaned up we'll figure something out, no reason to cause you pain now.